### PR TITLE
Add support for the uiSchema in react-jsonschema-form for plugins

### PR DESCRIPTION
**Pull request by sbender9**
Allows more customization of plugin configuration forms

### CODE REVIEW

**Comment by sbender9** [comment]
FYI, I ran prettier on plugins.js and it made a bunch of changes...

File: lib/interfaces/plugins.js [hunk 1]
```diff
@@ -14,153 +14,191 @@
...
+              console.log(e.code + " " + e.path);
+            }
+            var uiSchema = plugin.uiSchema;
+            if (
+              typeof uiSchema === "undefined" ||
+              typeof uiSchema["configuration"] === "undefined" ||
+              typeof uiSchema["configuration"]['ui:order'] === "undefined"
+            ) {
+              var order = [];
+              Object.keys(plugin.schema.properties).forEach(name => {
```
Comment (tkurki) [inline lib/interfaces/plugins.js]: This generates ui:order for schema's top level properties, but not for nested levels. Is this as it should be?

**Comment by tkurki** [comment]
With uiSchema in place some custom widgets would probably make sense for some config options. For example a custom widget for picking a SK path with autocomplete and/or validation of the value.

File: lib/interfaces/plugins.js (hunk 1 above)
Comment (sbender9) [reply lib/interfaces/plugins.js]: No, it's not how it should be. We should not be generating ui:order at all. Plugin developers should do that, But I was looking at making a bunch of pull requests to everyone else's plugins to add ordering. Or make something that works by default for most cases...

Comment (tkurki) [reply lib/interfaces/plugins.js]: Either make this work properly for nested properties or scrap it - I don't think it makes sense to have it work for only top level properties.

And the reasoning for this ("json order is not guaranteed and on ios is not preserved in practice") should be included in the commit description or in PR.

**Comment by tkurki** [comment]
Re: prettier: I try to make formatting changes in a separate commit. Makes reviewing stuff  easier, when you can separate the functional changes from formatting ones. Not possible always.

File: lib/interfaces/plugins.js (hunk 1 above)
Comment (sbender9) [reply lib/interfaces/plugins.js]: ok

**Comment by sbender9** [comment]
Starting over...

**Pull request closed without merging**